corehq/motech/generic_inbound/core.py

```diff
@@ -16,6 +19,9 @@ def execute_generic_api(domain, couch_user, device_id, context, api_model):
         # the bulk API always requires a list
         data = [data]
 
+    if not all(isinstance(item, dict) for item in data):
+        raise GenericInboundApiError(_("Unexpected type for transformed request"))
```

Comment:
ha, I added a similar check on a local branch I've been working on - noticed that `data` will be `None` if there's a problem parsing the expression.